Traceback lines on regex not found

### Version Number

2.2.2-build20

### What branch are you on?

nightly

### Describe the Bug

Maybe this is an FR or keep it as a Bug:

Gracefully exit with an error and less of this "scary" Traceback information over several lines. This is happening on the defaults for overlays: `- default: resolution`


`[2026-01-16 07:04:53,994] [overlays.py:625]           [DEBUG]    | Traceback (most recent call last):                                                                 |
                                                                 |   File "C:\Users\(redacted)\Quickstart\config\kometa\modules\overlays.py", line 582, in compile_overlays |
                                                                 |     builder = CollectionBuilder(self.config, overlay_file, k, v, library=self.library, overlay=True) |
                                                                 |               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ |
                                                                 |   File "C:\Users\(redacted)\Quickstart\config\kometa\modules\builder.py", line 1084, in __init__   |
                                                                 |     self._plex(method_name, method_data)                                                           |
                                                                 |   File "C:\Users\(redacted)\Quickstart\config\kometa\modules\builder.py", line 1993, in _plex      |
                                                                 |     self.builders.append((method_name, self.build_filter("plex_search", dict_data)))               |
                                                                 |                                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^                 |
                                                                 |   File "C:\Users\(redacted)\Quickstart\config\kometa\modules\builder.py", line 2707, in build_filter |
                                                                 |     built_filter, filter_text = _filter(base_dict, is_all=base_all)                                |
                                                                 |                                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^                                |
                                                                 |   File "C:\Users\(redacted)\Quickstart\config\kometa\modules\builder.py", line 2641, in _filter    |
                                                                 |     inside_filter, inside_display = _filter(dict_data, is_all=attr == "all", level=level)          |
                                                                 |                                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^          |
                                                                 |   File "C:\Users\(redacted)\Quickstart\config\kometa\modules\builder.py", line 2646, in _filter    |
                                                                 |     validation = self.validate_attribute(attr, modifier, final_attr, _data, validate, plex_search=True) |
                                                                 |                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ |
                                                                 |   File "C:\Users\(redacted)\Quickstart\config\kometa\modules\builder.py", line 2741, in validate_attribute |
                                                                 |     raise Failed(error)                                                                            |
                                                                 | modules.util.Failed: Plex Error: resolution: No matches found with regex pattern (?i)720|hd        |
                                                                 | Options: [('4K', '4k'), ('1080p', '1080')]                                                         |
                                                                 |                                                                                                    |
[2026-01-16 07:04:54,002] [overlays.py:626]           [ERROR]    | Plex Error: resolution: No matches found with regex pattern (?i)720|hd                             |
                                                                 | Options: [('4K', '4k'), ('1080p', '1080')]                                                         |
`

Here is another example related to a different builder "plex_search":

<img width="907" height="703" alt="Image" src="https://github.com/user-attachments/assets/c9863656-d6a0-48df-a4bb-b41e39ed83b3" />

### Relevant Collection/Overlay/Playlist Definition

```yml
#==================== Overlays ====================#
    overlay_files:
    - default: ribbon
    - default: mediastinger
    - default: ratings
      template_variables:
        rating1: user
        rating1_image: rt_tomato
        rating2: critic
        rating2_image: imdb
        rating3: audience
        rating3_image: tmdb
    - default: content_rating_us_movie
    - default: aspect
      template_variables:
        font_color: '#F5F6F7FF'
        stroke_width: 2
        stroke_color: '#E51515D9'
    - default: audio_codec
    - default: language_count
    - default: languages
    - default: languages
      template_variables:
        use_subtitles: true
    - default: resolution       #<-------HERE
    - default: runtimes
      template_variables:
        horizontal_offset: 392
        vertical_offset: 502
    - default: versions
    - default: video_format
    - default: streaming
    - default: studio
    - default: direct_play
      template_variables:
        horizontal_offset: -323
        vertical_offset: 657
```

### Logs

_No response_

--- a/.github/.wordlist.txt
+++ b/.github/.wordlist.txt
@@ -409,6 +409,7 @@ tomatoesaudience
 tomatometer
 tooltip
 tooltips
+traceback
 trakt
 trakt's
 transcoded

--- a/CHANGELOG
+++ b/CHANGELOG
@@ -86,3 +86,4 @@ Fix `mass_imdb_parental_labels` crash when IMDb GraphQL returns `null` for the t
 Addresses edge case where missing `settings:minimum_items` would make Kometa redact `1` in the log.
 Update Letterbox URL parsing to handle unlisted lists and shuffle argument in the url
 Fix an issue where tvdb_to_imdb writes wrong IMDb ID when TVDb movie ID collides with a TVDb series ID
+Hide some traceback errors for "known good" errors that don't really need a trace

--- a/modules/logs.py
+++ b/modules/logs.py
@@ -25,6 +25,12 @@ INFO = 20
 DEBUG = 10
 TRACE = 0
 
+SUPPRESS_STACKTRACE_PATTERNS = [
+    r"Plex Error: .* not found",
+    r"No matches found with regex pattern",
+    r"No Items found in Plex",
+]
+
 
 def fmt_filter(record):
     record.levelname = f"[{record.levelname}]"
@@ -218,7 +224,17 @@ class MyLogger:
             self._log(CRITICAL, str(msg), args, **kwargs)
 
     def stacktrace(self, trace=False):
-        self.print(traceback.format_exc(), debug=not trace, trace=trace)
+        stack = traceback.format_exc()
+
+        suppress_stacktrace_patterns = [
+            r"Plex Error: .* not found",
+            r"No matches found with regex pattern",
+        ]
+
+        if any(re.search(pattern, stack) for pattern in suppress_stacktrace_patterns):
+            return
+
+        self.print(stack, debug=not trace, trace=trace)
 
     def _space(self, display_title):
         display_title = str(display_title)

--- a/modules/logs.py
+++ b/modules/logs.py
@@ -229,6 +229,7 @@ class MyLogger:
         suppress_stacktrace_patterns = [
             r"Plex Error: .* not found",
             r"No matches found with regex pattern",
+            r"Plex Error: No Items found in Plex"
         ]
 
         if any(re.search(pattern, stack) for pattern in suppress_stacktrace_patterns):

